chia/server/introducer_peers.py

```diff
@@ -1,31 +1,52 @@
 import random
 import time
-from typing import Dict, List, Optional
+from typing import Set, List, Optional
+from dataclasses import dataclass
 
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.peer_info import PeerInfo
-from chia.util.ints import uint64
+from chia.util.ints import uint64, uint16
 
 
+@dataclass(frozen=False)
+class VettedPeer:
```

Comment:
this is the main change in this patch. Instead of using `PeerInfo` (which is frozen), use this type that also keeps the 3 extra members. This means we don't need to keep these in separate hash tables.

Reply:
looks good!